tests/pytests/unit/transport/test_ipc.py

```diff
@@ -34,3 +52,52 @@ async def test_ipc_connect_sync_wrapped(io_loop, tmp_path):
     with pytest.raises(tornado.iostream.StreamClosedError):
         # Don't `await subscriber.connect()`, that's the purpose of the SyncWrapper
         subscriber.connect()
+
+
+@pytest.fixture
+def master_config(sock_dir):
+    conf = salt.config.master_config("")
+    conf["sock_dir"] = sock_dir
+    yield conf
+
+
+@pytest.mark.skip_on_windows(reason="Unix socket not available on win32")
+def test_master_ipc_server_unix(master_config, sock_dir):
+    assert master_config.get("ipc_mode") != "tcp"
+    server = salt.transport.ipc_publish_server("master", master_config)
+    assert server.pub_path == str(sock_dir / "master_event_pub.ipc")
```

Comment:
Given the above comments about the existing master and minion config fixtures, maybe something like this would help here.

```python
@pytest.fixture
def sock_dir(tmp_path):
    sock_dir_path = tmp_path / "test-socks"
    sock_dir_path.mkdir(parents=True, exist_ok=True)
    yield sock_dir_path


@pytest.fixture
def minion_config(sock_dir, minion_opts):
    minion_opts["sock_dir"] = sock_dir
    yield minion_opts


@pytest.fixture
def master_config(sock_dir, master_opts):
    master_opts["sock_dir"] = sock_dir
    yield master_opts